testcases/kernel/syscalls/ioctl/ioctl07.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2017 Carlo Marcelo Arenas Belon <carenas@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+ * Very basic test for the RND* ioctls.
+ *
+ * Reads the entropy available from both /proc and the ioctl and compares
+ * they are similar enough (within a configured fuzz factor)
+ *
+ */
+
+#include <linux/random.h>
+#include <stdlib.h>
+#include "tst_safe_stdio.h"
+#include "tst_test.h"
+
+static char *s_fuzz;
+static int fuzz = 2;
+static struct tst_option options[] = {
+	{"f:", &s_fuzz, "-f c     Fuzz factor to considered valid (default 2)"},
+	{NULL, NULL, NULL}
+};
+
+static int fd;
+
+static void verify_ioctl(void)
+{
+	int cnt, pcnt;
+
+	FILE *proc = SAFE_FOPEN("/proc/sys/kernel/random/entropy_avail", "r");
+
+	SAFE_IOCTL(fd, RNDGETENTCNT, &cnt);
+	if (fscanf(proc, "%d", &pcnt) != 1)
+		tst_res(TBROK | TERRNO, "/proc failure");
+	SAFE_FCLOSE(proc);
+	tst_res(TINFO, "RND original value %d", cnt);
+
+	if (abs(pcnt - cnt) <= fuzz)
+		tst_res(TPASS, "entropy value from ioctl as expected");
+	else
+		tst_res(TFAIL, "incorrect entropy value from ioctl");
+}
+
+static void setup(void)
+{
+	fd = SAFE_OPEN("/dev/urandom", O_RDONLY);
+	if (s_fuzz)
+		fuzz = SAFE_STRTOL(s_fuzz, 0, 4096);
+}
+
+static void cleanup(void)
+{
+	if (fd > 0)
+		SAFE_CLOSE(fd);
+}
+
+static struct tst_test test = {
+	.needs_root = 0,
+	.needs_device = 0,
```

Comment:
static variables without explicit initializers are initialized to zero. So there is no need to set anything to zero in the tst_test structure.

Reply:
my bad, leftover from original copy/paste, corrected now

wanted to keep needs_root visible so it is clear it needs to be updated when all other RND* that require root are implemented